src/nats.cr

```diff
@@ -488,21 +486,25 @@ module NATS
             bytes += key.bytesize + value.bytesize + 4 # 2 extra bytes for ": " and 2 for CR+LF
           end
           @io << ' ' << header_length
-          @io << ' ' << header_length + message.bytesize << "\r\n"
+          @io << ' ' << header_length + payload.bytesize << "\r\n"
           @io << nats_header_preamble
           headers.each do |key, value|
             @io << key << ": " << value << "\r\n"
           end
           @io << "\r\n"
         else
-          @io << ' ' << message.bytesize << "\r\n"
+          @io << ' ' << payload.bytesize << "\r\n"
         end
 
-        @io.write message.to_slice
+        @io.write payload.to_slice
         @io << "\r\n"
       end
     end
 
+    def publish(message : Message) : Nil
+      publish message.subject, message.payload, message.reply_to, message.headers
+    end
```

Comment:
I'm making NATS a more central part of a service I'm working on and its nice to pass around NATS::Message or return new ones from functions so that other functions can then send in a reply or publish. Some encapsulation.

Reply:
Yeah, I don't mind this change. It seems all the first-party clients have this form of this method. The Go client seems to _only_ have this form, even.